cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/ClientConfig.java

```
@@ -109,6 +134,50 @@ private ClientConfig(Map<String, String> options)
     }
 
+    private ClearSnapshotStrategy parseClearSnapshotStrategy(boolean hasDeprecatedOption,
+                                                             boolean clearSnapshot,
+                                                             String clearSnapshotStrategyOption)
+    {
+        if (hasDeprecatedOption)
+        {
+            LOGGER.warn("The deprecated option 'clearSnapshot' is set. Please set 'clearSnapshotStrategy' instead.");
+            if (clearSnapshotStrategyOption == null)
+            {
+                return clearSnapshot ? ClearSnapshotStrategy.defaultStrategy() : new ClearSnapshotStrategy.NoOp();
+            }
+        }
+        if (clearSnapshotStrategyOption == null)
+        {
+            LOGGER.debug("No clearSnapshotStrategy is set. Using the default strategy");
+            return ClearSnapshotStrategy.defaultStrategy();
+        }
+        String[] strategyParts = clearSnapshotStrategyOption.split(" ");
+        String strategyName;
+        String snapshotTTL = null;
+        if (strategyParts.length == 1)
+        {
+            strategyName = strategyParts[0].trim();
+        }
+        else if (strategyParts.length == 2)
+        {
+            strategyName = strategyParts[0].trim();
+            snapshotTTL = strategyParts[1].trim();
+            if (!Pattern.matches(SNAPSHOT_TTL_PATTERN, snapshotTTL))
+            {
+                String msg = "Incorrect value set for clearSnapshotStrategy, expected format is " +
+                             "{strategy [snapshotTTLvalue]}. TTL value specified must contain unit along. " +
+                             "For e.g. 2d represents a TTL for 2 days";
+                throw new IllegalArgumentException(msg);
+            }
+        }
+        else
+        {
+            LOGGER.error("Invalid value for ClearSnapshotStrategy: '{}'", clearSnapshotStrategyOption);
+            throw new IllegalArgumentException("Invalid value: " + clearSnapshotStrategyOption);
```

In ClearSnapshotStrategy.create the fallback behavior is to return the default strategy. Should we consider that fallback strategy here as well?

---

Should we fail fast there too?. If users are setting `clearSnapshotStrategy` then they might want a particular behaviour, if that behaviour can't be met, better to alert earlier.

---

Fail fast sounds good.